app/kubernetes-ingress-controller/1.0.x/concepts/ingress-versions.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Ingress v1 and v1beta1 differences
+---
+
+## Introduction
+
+Kubernetes 1.18 introduced a new v1 API for the [Ingress resource][kubernetes-ingress-doc].
```

Comment:
AFAIK `networking.k8s.io/v1` is available by default as of 1.19, not 1.18

Reply:
Also, `v1beta1` is ambiguous: it can be either `extensions/v1beta1` or ``extensions.k8s.io/v1beta1`. It's not unusual for apiversions to be moved between groups like that.

Reply:
1.18 introduced it although it was a beta feature--it wasn't novel in 1.19, just graduated to GA.

The v1beta1 ambiguity is intentional here, since it exists in either extensions or networking depending on which K8S version you use, though its functionality is identical. That history isn't the focus here, and IMO the shorthand is fine for the differences this document seeks to cover.

Reply:
With `networking.k8s.io/v1beta1` explicitly called out in the document, I'm on board :+1:

Reply:
On 1.18 vs 1.19: with the current phrasing, I am expecting users coming to us with a complaint that "your docs said that I can use ingressv1 with 1.18, I tried it and I failed"

> I can't test that in our GKE cluster, as GKE 1.18 still rejects v1 Ingresses and 1.19 isn't yet available

this experience seems to indicate that confusion here (about the feature being opt-in) is a real thing.

Reply:
Fair enough. Changed to 1.19 on the assumption that beta users are savvy enough to understand the distinction and know that they've enabled the gate, and that less savvy users will just use 1.19.